src-electron/generator/helper-endpointconfig.js

```diff
@@ -295,20 +295,39 @@ function endpoint_attribute_list(options) {
   return ret
 }
 
-function endpoint_fixed_device_id_array(options) {
-  return (
-    '{ ' + this.deviceList.map((device) => device.deviceId).join(', ') + ' }'
-  )
-}
-
-function endpoint_fixed_device_version_array(options) {
+function endpoint_fixed_device_type_array(options) {
   return (
     '{ ' +
-    this.deviceList.map((device) => device.deviceVersion).join(', ') +
+    this.deviceList
+      .map(function (device) {
+        console.log(device)
+        if (device != null && device.deviceId != null) {
+          return (
+            '{ ' +
+            device.deviceId.toString() +
+            ',' +
+            device.deviceVersion.toString() +
+            '}'
+          )
+        } else {
+          return '{}'
```

Comment:
What will this init things to in practice?

Reply:
An `EmberAfDefinedEndpoint` that has a `EmberAfDeviceType` instance that points to a null list.

Reply:
This initializes to just a default constructed `EmberAfDeviceType`, which is actually not what we want. What we do want is to make it such that the appropriate length for that endpoint is set to 0.